src/main/java/com/softserve/academy/spaced/repetition/service/UserService.java

```diff
@@ -200,22 +207,34 @@ public User uploadImage(MultipartFile file) throws ImageRepositorySizeQuotaExcee
             NotAuthorisedUserException, FileIsNotAnImageException {
         imageService.checkImageExtention(file);
         User user = getAuthorizedUser();
-        user.getPerson().setImageBase64( imageService.encodeToBase64(file));
+        user.getPerson().setImageBase64(imageService.encodeToBase64(file));
         user.getPerson().setTypeImage(ImageType.BASE64);
         return userRepository.save(user);
     }
 
-    public byte[] getDecodedImageContent() throws NotAuthorisedUserException{
+    public byte[] getDecodedImageContent() throws NotAuthorisedUserException {
         User user = getAuthorizedUser();
         String encodedFileContent = user.getPerson().getImageBase64();
         return imageService.decodeFromBase64(encodedFileContent);
     }
 
-
-    public void deleteAccount() throws NotAuthorisedUserException{
+    public void deleteAccount() throws NotAuthorisedUserException {
         User user = getAuthorizedUser();
         user.getAccount().setStatus(AccountStatus.INACTIVE);
         userRepository.save(user);
         mailService.sendAccountNotificationMail(user);
     }
+
+    public void getUserStatus() throws UserIsDeletedException, UserIsBlockedException, UserIsInactiveException {
+        JwtUser jwtUser = (JwtUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        User user = userRepository.findUserByAccountEmail(jwtUser.getUsername());
+        if (user.getAccount().getStatus().equals(AccountStatus.DELETED)) {
```

Comment:
Instead of introducing three new exception classes:
1. Introduce 1 exception class, e.g. 
`public class UserStatusException extends Exception {`
2. Make it storing the status by introducing constructor
`public UserStatusException(AccountStatus status) {`
3. Let the ExceptionHandler decidewhich status code and message to return dependingon the status.

Reply:
This method then should look like 
```
if (user.getAccount().getStatus().isNotActive()) {
    throw new UserStatusException(user.getAccount().getStatus());
}
```